src/fetch/FetchExample.re

```diff
@@ -0,0 +1,93 @@
+type tDog = string;
+
+type tArrayDogs = array(tDog);
+
+type state = { dogs: tArrayDogs, isLoading: bool, error: bool };
+
+type action =
+  | DogFetch
+  | DogFetched(tArrayDogs)
+  | DogFailedToFetch;
+
+let component = ReasonReact.reducerComponent("FetchExample");
```

Comment:
You should not put anything in between `let component` and `let make` since that can cause some hard ot understand type inference errors.